rpm/0035-Add-OriginalSubject-message-sort-key.patch

```diff
@@ -0,0 +1,119 @@
++    static QMailMessageSortKey originalSubject(Qt::SortOrder order = Qt::AscendingOrder);
+     static QMailMessageSortKey timeStamp(Qt::SortOrder order = Qt::AscendingOrder);
+     static QMailMessageSortKey receptionTimeStamp(Qt::SortOrder order = Qt::AscendingOrder);
+     static QMailMessageSortKey serverUid(Qt::SortOrder order = Qt::AscendingOrder);
+diff --git a/src/libraries/qmfclient/qmailstore_p.cpp b/src/libraries/qmfclient/qmailstore_p.cpp
+index 97f5020d..bc077f7c 100644
+--- a/src/libraries/qmfclient/qmailstore_p.cpp
++++ b/src/libraries/qmfclient/qmailstore_p.cpp
+@@ -632,6 +632,7 @@ static QMap<QMailMessageSortKey::Property, QMailMessageKey::Property> messageSor
+     map.insert(QMailMessageSortKey::Sender, QMailMessageKey::Sender);
+     map.insert(QMailMessageSortKey::Recipients, QMailMessageKey::Recipients);
+     map.insert(QMailMessageSortKey::Subject, QMailMessageKey::Subject);
++    map.insert(QMailMessageSortKey::OriginalSubject, QMailMessageKey::TimeStamp);
+     map.insert(QMailMessageSortKey::TimeStamp, QMailMessageKey::TimeStamp);
+     map.insert(QMailMessageSortKey::ReceptionTimeStamp, QMailMessageKey::ReceptionTimeStamp);
+     map.insert(QMailMessageSortKey::Status, QMailMessageKey::Status);
+@@ -8572,7 +8573,19 @@ QMailStorePrivate::AttemptResult QMailStorePrivate::attemptQueryMessages(const Q
+                                                                          QMailMessageIdList *ids, 
+                                                                          ReadLock &)
+ {
+-    QSqlQuery query(simpleQuery(QLatin1String("SELECT id FROM mailmessages"),
++    bool sortByOriginalSubject = false;
++    bool descendingOrder = false;
++    for (const QMailMessageSortKey::ArgumentType &arg : sortKey.arguments()) {
++        if (arg.property == QMailMessageSortKey::OriginalSubject) {
++            sortByOriginalSubject = true;
++            descendingOrder = arg.order == Qt::DescendingOrder;
++            break;
++        }
++    }
++
++    QSqlQuery query(simpleQuery(sortByOriginalSubject
++                                ? QLatin1String("SELECT id, subject FROM mailmessages")
++                                : QLatin1String("SELECT id FROM mailmessages"),
+                                 QVariantList(),
+                                 QList<Key>() << Key(key) << Key(sortKey),
+                                 qMakePair(limit, offset),
+@@ -8580,8 +8593,23 @@ QMailStorePrivate::AttemptResult QMailStorePrivate::attemptQueryMessages(const Q
+     if (query.lastError().type() != QSqlError::NoError)
+         return DatabaseFailure;
+ 
+-    while (query.next())
+-        ids->append(QMailMessageId(extractValue<quint64>(query.value(0))));
++    if (sortByOriginalSubject) {
++        QMap<QString, quint64> map;
++        while (query.next())
++            map.insertMulti(extractValue<QString>(query.value(1)).replace(QRegExp(QStringLiteral("^(re:|fw:|fwd:|\\s*|\\\")*"), Qt::CaseInsensitive), QString()),
```

Comment:
Suppose it would be better creating the regexp outside the loop, i.e. not separately for every result.

Also on new code it could be better using QRegularExpression instead of QRegexp.